File: Back/src/main/java/com/gestion/adhesion/models/Activite.java
package com.gestion.adhesion.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

import static javax.persistence.GenerationType.SEQUENCE;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@EqualsAndHashCode(of = {"id"})
@Entity
@Table(	name = "activites")
@JsonIgnoreProperties(ignoreUnknown = true)
public class Activite {

    @Id
    @SequenceGenerator(
            name = "activite_sequence",
            sequenceName = "activite_sequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = SEQUENCE,
            generator = "activite_sequence"
    )
    private Long id;

    private String groupe;

    private String nom;
    private String lien;
    private Integer tarif;

    private Integer nbPlaces;

    private String horaire;

    private String salle;

    private Boolean pourEnfant;

    private boolean priseEnCharge;

    private boolean autorisationParentale;

    private boolean vieClub;

    private boolean certificatMedical;

    private Integer dureeVieCertif;

    @Transient
    private int nbAdherentTotal;

    @OneToMany(mappedBy="activite", cascade = CascadeType.ALL)
    @JsonIgnore
    @ToString.Exclude
    private Set<Adhesion> adhesions = new HashSet<>();

    @ManyToMany(mappedBy="cours", cascade = CascadeType.ALL)
    @JsonIgnoreProperties({"cours", "accords", "adhesions", "user", "tribu"})
    private Set<Adherent> profs = new HashSet<>();
}

File: client/src/app/models/activite.ts
import { Adherent } from "./adherent";
import { Adhesion } from "./adhesion";

export class Activite {

  id!: number;
  nom!: string;
  groupe!: string;
  groupeFiltre!: string;
  tarif!: number;

  ageMin!: number;
  ageMax!: number;
  genre!: string;
  horaire!: string;
  lien!: string;
  salle!: string;
  profs!: Adherent[];
  nbPlaces!: number;
  nbAdhesionsEnCours!: number;
  nbAdhesionsCompletes!: number;
  nbAdhesionsAttente!: number;
  montantCollecte!: number;
  adhesions!: Adhesion[];
  priseEnCharge!: boolean;
  autorisationParentale!: boolean;
  certificatMedical!: boolean;
  vieClub!: boolean;
  complete!: boolean;
  reinscription!: boolean;

  dureeVieCertif!: number;

  constructor() {

    this.profs = [];
    this.adhesions = [];

    this.nom = 'nouvelle activité'
    this.groupeFiltre = 'Amicaliste'
    this.horaire = 'dimanche matin'
    this.groupe = 'ALOD_G'
    this.tarif = 200
    
    this.nbPlaces = 10;
    this.ageMin = 16;
    this.ageMax = 99;
    this.genre = 'Non genrée';
    this.priseEnCharge = false;
    this.autorisationParentale = false;
    this.certificatMedical = false;
    this.vieClub = false;
    this.complete = false;
    this.reinscription = false;

  }
}
